src/util.h

```diff
@@ -117,6 +117,20 @@ namespace rhost {
             }
         };
 
+        struct blob {
```

Comment:
This is effectively a view of a slice of a blob, rather than a blob (i.e. it doesn't manage the lifetime of the data, and is not valid as soon as the buffer into which `raw_data` points goes away). It probably deserves a more descriptive name that would make it clear - something like `blob_slice` - and some comments to that effect.